core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -3101,7 +3101,7 @@ public void heartbeat() {
               LOG.debug("File {} is expired. Performing action {}", inode.getName(), ttlAction);
               switch (ttlAction) {
                 case FREE:
-                  free(path, FreeOptions.defaults().setForced(true));
+                  free(path, FreeOptions.defaults().setForced(true).setRecursive(true));
```

Comment:
nit: would it make sense to change the `DELETE` case to also always use recursive == true for consistency?